src/rust-crypto/keybackup.ts

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { ISigned } from "../@types/signed";
+
+export interface Curve25519AuthData {
```

Comment:
these public classes belong in `src/crypto-api/` (which is public), not `src/rust-crypto` (which is internal).

You can move the whole file into `src/crypto-api/` but please remember to add an `export *` to `src/crypto-api.ts`  so that they are properly exported as public identifiers from `matrix-js-sdk`.